tests/ludwig/modules/test_metric_modules.py

```diff
@@ -36,12 +37,17 @@ def test_rmspe_metric(preds: torch.Tensor, target: torch.Tensor, output: torch.T
     [
         (torch.arange(3), torch.arange(3, 6), 1, torch.tensor(-12.5)),
         (torch.arange(6).reshape(3, 2), torch.arange(6, 12).reshape(3, 2), 2, torch.tensor(-12.5)),
+        (torch.tensor([0.8]), torch.arange(1), 1, np.nan),
```

Comment:
Instead of overloading this test, and adding additional conditional logic in the test to handle this test case, I would create a separate test (unparameterized) for this special sample=1 case, i.e. `test_r2_score_single_prediction`.